plugins/linux-v4l2/v4l2-output.c

```diff
@@ -6,7 +6,8 @@
 #include <fcntl.h>
 #include <unistd.h>
 
-#define MAX_DEVICES 64
+#define __USE_GNU
```

Comment:
Is this really correct / okay to do here? According to [this old email](https://gcc.gnu.org/legacy-ml/fortran/2005-10/msg00365.html), `__USE_GNU` is not supposed to be defined by applications - apparently `_GNU_SOURCE` is what should be defined.

Reply:
I had a feeling about that. I guess i simply followed the man page or peeked header file. Also I believe I tried `_GNU_SOURCE` before but didn't work on first try. I think it really just needed to be on the very top of the `#includes`, thanks.

Reply:
Would you mind explaining (in the commit message) why defining `_GNU_SOURCE` is necessary? Actually, does this commit work if you just don't define it?

Reply:
Yes it is necessary.
```C
n = scandir("/dev", &list, scanfilter, versionsort);
```
Here `versionsort` is a callback function that is a GNU extension.

Any suggestion for a log message wording?

Reply:
Perhaps something like:

```
Since versionsort() is a GNU extension, define _GNU_SOURCE at
the top of the source file.
```

Reply:
My brain came up with something similar. I hope that fits.

Reply:
Thank you!